### PR TITLE
MM-15481 system_permission_gate migration to typescript

### PR DESCRIPTION
**Pull request by mathiusjohnson**
#### Summary
Migrated system_permission_gate to typescript. 
index.js migrated to index.ts
system_permission_gate.test.jsx now system_permission_gate.test.tsx
system_permision_gate.jsx now system_permision_gate.tsx
changed propTypes to type with appropriate types declared for properties
SystemPermissionGate class has public static defaultProps

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15481

### CODE REVIEW

File: components/permissions_gates/system_permission_gate/system_permission_gate.tsx [hunk 1]
```diff
@@ -1,40 +1,35 @@
...
-        invert: false,
-    }
-
-    static propTypes = {
-
-        /**
-         * Permissions enough to pass the gate (binary OR)
-         */
-        permissions: PropTypes.arrayOf(PropTypes.string).isRequired,
+    type Props = {
```
Comment (devinbinnie) [inline components/permissions_gates/system_permission_gate/system_permission_gate.tsx:6]: Nit: fix indenting

File: components/post_view/post_info/post_info.test.jsx [hunk 2]
```diff
@@ -5,8 +5,9 @@ import React from 'react';
 import {shallow} from 'enzyme';
 import {Posts} from 'mattermost-redux/constants';
 
+import PostInfo from 'components/post_view/post_info/post_info';
```
Comment (devinbinnie) [inline components/post_view/post_info/post_info.test.jsx:8]: This change seems unrelated to the component you migrated. How come it ended up in here?
It's a good change so it's fine, just curious.

**Review by devinbinnie — APPROVED** [review]
LGTM! Just have a couple non-blocking comments.
Thanks @mathiusjohnson!

**Review by calebroseland — APPROVED** [review]
LGTM ✓

**Review by stevemudie — APPROVED** [review]
LGTM

File: components/post_view/post_info/post_info.test.jsx (hunk 2 above)
Comment (mathiusjohnson) [reply components/post_view/post_info/post_info.test.jsx:8]: My bad, I started to work on post_info migration but decided to begin with something simpler :) must have missed that undo!
